src/mlpack/methods/ann/dists/normal_distribution.hpp

```diff
@@ -0,0 +1,135 @@
+/**
+ * @file normal_distribution.hpp
+ * @author Atharva Khandait
+ *
+ * Definition of the Normal distribution class.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_DISTRIBUTIONS_NORMAL_DISTRIBUTION_HPP
+#define MLPACK_METHODS_ANN_DISTRIBUTIONS_NORMAL_DISTRIBUTION_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * Multiple univariate Normal(Gaussian) distributions.
+ *
+ * @tparam DataType Type of the input data. (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ */
+template <typename DataType = arma::mat>
```

Comment:
Sorry, I'm a bit confused by this one, wouldn't this class only be used where the data type is an `arma::vec`?  (That is, a vector with one element for each dimension)

Reply:
As this is a class allowing multiple univariate normal distributions, it can be any data type. For example, two 5 x 5 matrices with means in one and standard deviations in the other. This will have 25 univariate distributions with each element of a collective sample matrix denoting a sample from the corresponding parameters.

Reply:
Okay, if it’s just changing the default template parameter then it’s not a problem. If I move it to core, then I will change it to vector.

Reply:
Is it okay if I keep it `arma::mat` as it will be in the ann module?

Reply:
Right, since it will stay in the ann folder, I see no need to make the change and my comment here can be ignored. :)